### DISCUSSION TITLE
Several segfaults since nightly SSL fix deployed

### DISCUSSION THREAD

**Discussion by dizorg** [netdata/netdata#13211] [Q&A]
```
[Thu Jun 16 15:51:38 2022] netdata[15865]: segfault at 0 ip 000000000047ee77 sp 00007fffaa355ea0 error 4 in netdata[400000+234000]
[Thu Jun 16 15:56:53 2022] netdata[18309]: segfault at 0 ip 000000000047ee77 sp 00007ffe86674ef0 error 4 in netdata[400000+234000]
[Thu Jun 16 16:00:22 2022] netdata[27971]: segfault at 0 ip 000000000047ee77 sp 00007fff9d02d590 error 4 in netdata[400000+234000]
[Thu Jun 16 16:04:51 2022] netdata[29426]: segfault at 0 ip 000000000047ee77 sp 00007ffcfe7c1ce0 error 4 in netdata[400000+234000]
[Thu Jun 16 16:07:56 2022] netdata[31838]: segfault at 0 ip 000000000047ee77 sp 00007ffccd9353e0 error 4 in netdata[400000+234000]
[Fri Jun 17 00:29:18 2022] SERVICE[2103]: segfault at 0 ip 000000000047ec54 sp 00007f23090ffb20 error 4 in netdata[400000+234000]
[Mon Jun 20 08:04:32 2022] SERVICE[6545]: segfault at 0 ip 000000000047ec54 sp 00007fafcc69ab20 error 4 in netdata[400000+234000]
[Tue Jun 21 08:47:54 2022] SERVICE[24499]: segfault at 0 ip 000000000047ec54 sp 00007f1c11464b20 error 4 in netdata[400000+234000]
[Wed Jun 22 17:12:44 2022] SERVICE[5066]: segfault at 0 ip 000000000047ec54 sp 00007f5663f36b20 error 4 in netdata[400000+234000]
[Thu Jun 23 14:29:33 2022] netdata[28304]: segfault at 0 ip 000000000047ee77 sp 00007ffec0514b80 error 4 in netdata[400000+234000]
[Thu Jun 23 14:31:03 2022] netdata[29325]: segfault at 0 ip 000000000047cee3 sp 00007ffdc552b670 error 4 in netdata[400000+234000]
```

Anyone else seeing this? Channel nightly-1.35.0 after SSL 100% CPU fix.

**Comment by MrZammler** [comment]
Hi @dizorg ! Is it possible to capture a core dump? Would most likely need to build with debugging information. Can you elaborate more on your setup? OS, ssl version, number of children, etc. ?

**Reply by thiagoftsm** [reply]
Hello @dizorg ,

Please, also give us the output of the command `openssl version`.

Best regards!

**Reply by dizorg** [reply]
$ openssl version
OpenSSL 1.0.2k-fips  26 Jan 2017

**Reply by thiagoftsm** [reply]
Thank you @dizorg, I have a CentOS 7 VM running  with `openssl-1.0.2k-25.el7_9.x86_64` and I will try to recreate the issue using it.

**Reply by thiagoftsm** [reply]
Hello @dizorg ,

Torday I ran netdata using the nightly and I tried to crash it as you described. I configured a `CentOS 7.x` virtual machine with a `TLS certificate` and the same machine was also streaming data for another host using TLS. I tested the dashboard using a [script](https://github.com/netdata/netdata/tree/master/tests/urls) to do thousand of request o Netdata's api. I could not crash netdata in this scenario, so I have more two questions for you:

1. What is the output of the next command?
```sh
[root@localhost urls]# openssl version -f
compiler: gcc -I. -I.. -I../include  -fPIC -DOPENSSL_PIC -DZLIB -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -DKRB5_MIT -m64 -DL_ENDIAN -Wall -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -Wa,--noexecstack -DPURIFY -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM
```
2. Which Linux distribution are you running?

Best regards!

**Reply by dizorg** [reply]
Hi @thiagoftsm - here are the details:

```
# cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

# openssl version -f
compiler: gcc -I. -I.. -I../include  -fPIC -DOPENSSL_PIC -DZLIB -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -DKRB5_MIT -m64 -DL_ENDIAN -Wall -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -Wa,--noexecstack -DPURIFY -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM
#
```

**Reply by dizorg** [reply]
I have about 40 children, not all in the same streaming protocol version, not all with SSL enabled. I was enabling SSL to tackle gaps in graphs of one of the children that is behind a netdata collector+proxy sending data to the parent instance.

**Reply by dizorg** [reply]
```
# cat /etc/centos-release
CentOS Linux release 7.9.2009 (Core)
```

**Reply by dizorg** [reply]
I have one question regarding the topology described: 

Child->Proxy-->Parent

I'm activating SSL only from Proxy to Parent, should I also activate SSL from Child to Proxy?

**Reply by MrZammler** [reply]
@dizorg Hi!

Have you maybe pinpointed the problem with this child specifically? I.e. if this child is not connected, do the crashes still occur on the parent?

Can you also post a part of the `error.log` around the time of the crashes?

**Reply by thiagoftsm** [reply]
Hello @dizorg ,

I took a look in all your answers and I also try another way to crash netdata, and I did not have success. I observed that your `openssl version -f` has the same output than mine, but there is one thing that called my attention and I would like t ask you. When I run the following command:

```sh
[root@localhost vagrant]# rpm -qa | grep openssl
openssl-devel-1.0.2k-25.el7_9.x86_64
openssl-libs-1.0.2k-25.el7_9.x86_64
xmlsec1-openssl-1.2.20-7.el7_4.x86_64
openssl-1.0.2k-25.el7_9.x86_64
[root@localhost vagrant]#
```

I do not have the `fips` suffix attached to my openssl packages. How did you install your OpenSSL?

Best regards!

**Reply by dizorg** [reply]
I didn't install it on my own, it's a Google Cloud image though. 

This is what comes back for openssl installed packages:

```
# rpm -qa | grep openssl
openssl11-libs-1.1.1k-3.el7.x86_64
openssl-libs-1.0.2k-25.el7_9.x86_64
openssl11-1.1.1k-3.el7.x86_64
openssl-1.0.2k-25.el7_9.x86_64
openssl-devel-1.0.2k-25.el7_9.x86_64
#
```

**Reply by dizorg** [reply]
@MrZammler I've disabled SSL on the Proxy netdata to have it not to crash anymore... it's stable since a few weeks now.

**Reply by thiagoftsm** [reply]
Hello @dizorg ,

The openssl version that you listed with `rpm -qa | grep openssl` does not match the previous version you gave [here](https://github.com/netdata/netdata/discussions/13211#discussioncomment-3047757).

It is possible google updated their package, and this is the motive I cannot recreate the issue :thinking: , I am using the default packages released with CentOS.

**Reply by dizorg** [reply]
I just ran again both commands: 

```
# openssl version; rpm -qa| egrep openssl ; date
OpenSSL 1.0.2k-fips  26 Jan 2017
openssl11-libs-1.1.1k-3.el7.x86_64
openssl-libs-1.0.2k-25.el7_9.x86_64
openssl11-1.1.1k-3.el7.x86_64
openssl-1.0.2k-25.el7_9.x86_64
openssl-devel-1.0.2k-25.el7_9.x86_64
Fri Jul  1 14:10:19 UTC 2022
#
```

**Reply by dizorg** [reply]
Is there a recommended version of openssl I should be looking into upgrading it to?

**Reply by thiagoftsm** [reply]
Hello, netdata should run with any possible openssl, I asked you, because I am trying to recreate the same setup that you have. Unfortunately with the VM I am using, I did not have the same OpenSSL available.
